packages/undo-manager/src/test/index.js

```diff
@@ -0,0 +1,257 @@
+/**
+ * Internal dependencies
+ */
+import { createUndoManager } from '../';
+
+describe( 'Undo Manager', () => {
+	it( 'stacks undo levels', () => {
+		const undo = createUndoManager();
+
+		undo.addRecord( [
+			{ id: '1', changes: { value: { from: undefined, to: 1 } } },
```

Comment:
Why do we need to track "from" and "to"?

Reply:
When you go down the history, you need to apply the "from" value as an edit and when you go up you need to apply the "to" value. Storing only a single value creates bugs when you keep track of a history of multiple properties. This is explained better in the older PR that actually fixed this issue. https://github.com/WordPress/gutenberg/pull/50911#issue-1724145251

Reply:
Cool, maybe this could be in some inline docs somewhere? Up to you